otherlibs/dynlink/byte/dynlink.ml

```diff
@@ -134,7 +134,7 @@ module Bytecode = struct
           if compunit.cu_debug = 0 then [| |]
           else begin
             seek_in ic compunit.cu_debug;
-            [| input_value ic |]
+            [| Compression.input_value ic |]
```

Comment:
```suggestion
            [| Compression.input_value ic : Instruct.debug_event list array |]
```
I think this is a good opportunity to add the missing type constraint.

Reply:
PS.  The correct form was
```
[| (Compression.input_value ic : Instruct.debug_event list) |]
```